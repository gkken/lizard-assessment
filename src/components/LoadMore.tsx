import ChevronDownIcon from '@heroicons/react/24/outline/ChevronDownIcon';

const LoadMore = ({
  handleLoadMore,
}: {
  handleLoadMore: React.MouseEventHandler<HTMLButtonElement>;
}) => {
  return (
    <div className="flex justify-center pb-10">
      <button onClick={handleLoadMore}>
        <div className="font-bold">Load More</div>
        <div className="flex justify-center">
          <ChevronDownIcon className="w-5 stroke-2" />
        </div>
      </button>
    </div>
  );
};

export default LoadMore;
