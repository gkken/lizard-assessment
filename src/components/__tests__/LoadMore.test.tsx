import { render, screen, fireEvent } from '@testing-library/react';
import LoadMore from 'components/LoadMore';
const mockedLoadMore = jest.fn();

describe('LoadMore', () => {
  test('Should match snapshot', () => {
    expect(
      render(<LoadMore handleLoadMore={mockedLoadMore} />)
    ).toMatchSnapshot();
  });

  test('Runs handleLoadMore function when clicked', () => {
    render(<LoadMore handleLoadMore={mockedLoadMore} />);

    const loadMore = screen.getByRole('button');
    fireEvent.click(loadMore);
    expect(mockedLoadMore).toHaveBeenCalled();
  });
});
