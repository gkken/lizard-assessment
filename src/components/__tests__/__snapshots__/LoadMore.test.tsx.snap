// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMore Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center pb-10"
      >
        <button>
          <div
            class="font-bold"
          >
            Load More
          </div>
          <div
            class="flex justify-center"
          >
            <svg
              aria-hidden="true"
              class="w-5 stroke-2"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.5 8.25l-7.5 7.5-7.5-7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center pb-10"
    >
      <button>
        <div
          class="font-bold"
        >
          Load More
        </div>
        <div
          class="flex justify-center"
        >
          <svg
            aria-hidden="true"
            class="w-5 stroke-2"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5 8.25l-7.5 7.5-7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
